tests/timing_util_test.py

```
@@ -0,0 +1,171 @@
+
+import unittest
+
+from perfkitbenchmarker import flags_validators
+from perfkitbenchmarker import timing_util
+
+
+class ValidateMeasurementsFlagTestCase(unittest.TestCase):
+  """Tests exercising ValidateMeasurementsFlag."""
+
+  def testEmptyList(self):
+    """Passing an empty list is not allowed."""
+    e = None
+    try:
+      timing_util.ValidateMeasurementsFlag([])
+    except flags_validators.Error as e:
+      pass
+    self.assertIsNotNone(e)
+    self.assertEqual(str(e), 'option --timing_measurements requires argument')
+
+  def testInvalidValue(self):
+    """Passing an unrecognized value is not allowed."""
+    e = None
+    try:
+      timing_util.ValidateMeasurementsFlag(['TEST'])
+    except flags_validators.Error as e:
+      pass
+    self.assertIsNotNone(e)
+    self.assertEqual(str(e), 'TEST: Invalid value for --timing_measurements')
+
+  def testNoneWithAnother(self):
+    """Passing NONE with another value is not allowed."""
+    e = None
+    try:
+      timing_util.ValidateMeasurementsFlag(['NONE', 'RUNTIMES'])
+    except flags_validators.Error as e:
+      pass
+    self.assertIsNotNone(e)
+    self.assertEqual(
+        str(e), 'NONE: Cannot combine with other --timing_measurements options')
+
+  def testValid(self):
+    """Test various valid combinations."""
+    validate = timing_util.ValidateMeasurementsFlag
+    e = None
+    try:
+      self.assertEqual(validate(['NONE']), True)
+      self.assertEqual(validate(['END_TO_END_RUNTIME']), True)
+      self.assertEqual(validate(['RUNTIMES']), True)
+      self.assertEqual(validate(['TIMESTAMPS']), True)
+      self.assertEqual(validate(['END_TO_END_RUNTIME', 'RUNTIMES']), True)
+      self.assertEqual(validate(['END_TO_END_RUNTIME', 'TIMESTAMPS']), True)
+      self.assertEqual(validate(['RUNTIMES', 'TIMESTAMPS']), True)
+      self.assertEqual(
+          validate(['END_TO_END_RUNTIME', 'RUNTIMES', 'TIMESTAMPS']),
+          True)
+    except flags_validators.Error as e:
```

Let this fail rather than catching the error.